types/store.go

```diff
@@ -80,18 +84,32 @@ type (
 	MemoryStoreKey    = types.MemoryStoreKey
 )
 
+// assertNoCommonPrefix will panic if there are two keys: k1 and k2 in keys, such that
+// k1 is a prefix of k2
+func assertNoPrefix(keys []string) {
+	sorted := make([]string, len(keys))
+	copy(sorted, keys)
+	sort.Strings(sorted)
+	for i, n := range sorted {
+		if i > 0 && strings.HasPrefix(n, sorted[i-1]) {
+			panic(fmt.Sprint("Potential key collision between KVStores:", n, " - ", sorted[i-1]))
+		}
+	}
```

Comment:
```suggestion
	for i := 1; i < len(sorted); i++ {
		if strings.HasPrefix(sorted[i], sorted[i-1]) {
			panic(fmt.Sprint("Potential key collision between KVStores:", sorted[i], " - ", sorted[i-1]))
		}
	}
```

imo this would be better so we can remove the `i > 0` check each iteration

Reply:
yes, thanks for suggestion